flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/data/RowDataProjection.java

```diff
@@ -166,7 +169,8 @@ public int getArity() {
 
   @Override
   public RowKind getRowKind() {
-    return rowData.getRowKind();
+    // rowData can be null for nested struct
```

Comment:
Could you elaborate a bit on this?

Reply:
Dug out this issue: https://github.com/apache/iceberg/issues/2738

If the nested struct is null, right now both StructProjection and RowDataProjection returns a Projection object wrapping a null struct. I actually think it is probably better to just return a null Projection object in this case. 

Let me create a separate issue to follow up on this and see if the community agrees with the change.

Reply:
https://github.com/apache/iceberg/issues/7507